components/slidingPanel/slidingPanel.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component, PropTypes } from 'react';
+import { findDOMNode } from 'react-dom';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+
+export default class SlidingPanel extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { hideOverlay: true, isOpen: false };
+
+    this.handleClickOverlay = this.handleClickOverlay.bind(this);
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleTransitionEnd = this.handleTransitionEnd.bind(this);
+  }
+
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.node),
+      PropTypes.node,
+    ]).isRequired,
+
+    /**
+     * When true, if the user clicks on the overaly, closes the panel.
+     */
+    closeOnOverlay: PropTypes.bool,
```

Comment:
For consistency's sake, can we use the same prop name as in the Modal? `closeOnOverlayClick`